File: MatrixMultiplication/src/main/java/MatrixMultiplication.java
import java.util.Random;

public class MatrixMultiplication {
    private int size;
    private double[][] a;
    private double[][] b;
    private double[][] c;

    public MatrixMultiplication(int size) {
        this.size = size;
        a = new double[size][size];
        b = new double[size][size];
        c = new double[size][size];
    }

    public void initializeMatrices() {
        Random random = new Random();
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                a[i][j] = random.nextDouble();
                b[i][j] = random.nextDouble();
                c[i][j] = 0.0;
            }
        }
    }

    public void multiply() {
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                for (int k = 0; k < size; k++) {
                    c[i][j] += a[i][k] * b[k][j];
                }
            }
        }
    }
}

File: matrixMultiplication/MatrixMultiplication.py
import random

class MatrixMultiplication:
    def __init__(self, size):
        self.size = size
        self.a = [[0.0 for _ in range(size)] for _ in range(size)]
        self.b = [[0.0 for _ in range(size)] for _ in range(size)]
        self.c = [[0.0 for _ in range(size)] for _ in range(size)]

    def initialize_matrices(self):
        for i in range(self.size):
            for j in range(self.size):
                self.a[i][j] = random.random()
                self.b[i][j] = random.random()
                self.c[i][j] = 0.0

    def multiply(self):
        for i in range(self.size):
            for j in range(self.size):
                for k in range(self.size):
                    self.c[i][j] += self.a[i][k] * self.b[k][j]
